src/plugins/dashboard/public/dashboard_api/get_dashboard_api.ts

```diff
@@ -7,47 +7,268 @@
+    getPanelReferences,
+    (refs: Reference[]) => references.push(...refs)
+  );
+  const dataLoadingManager = initializeDataLoadingManager(panelsManager.api.children$);
+  const dataViewsManager = initializeDataViewsManager(
+    controlGroupApi$,
+    panelsManager.api.children$
+  );
+  const unifiedSearchManager = initializeUnifiedSearchManager(
+    initialState,
+    controlGroupApi$,
+    dataLoadingManager.internalApi.waitForPanelsToLoad$,
+    () => unsavedChangesManager.internalApi.getLastSavedState(),
+    creationOptions
+  );
+  const settingsManager = initializeSettingsManager({
+    initialState,
+    setTimeRestore: unifiedSearchManager.internalApi.setTimeRestore,
+    timeRestore$: unifiedSearchManager.internalApi.timeRestore$,
+  });
+  const unsavedChangesManager = initializeUnsavedChangesManager({
+    anyMigrationRun: savedObjectResult?.anyMigrationRun ?? false,
+    creationOptions,
+    controlGroupApi$,
+    lastSavedState: omit(savedObjectResult?.dashboardInput, 'controlGroupInput') ?? {
+      ...DEFAULT_DASHBOARD_INPUT,
+    },
+    panelsManager,
+    savedObjectId$,
+    settingsManager,
+    viewModeManager,
+    unifiedSearchManager,
+  });
+  async function getState() {
+    const { panels, references: panelReferences } = await panelsManager.internalApi.getState();
+    const dashboardState: DashboardState = {
+      ...settingsManager.internalApi.getState(),
+      ...unifiedSearchManager.internalApi.getState(),
+      panels,
+      viewMode: viewModeManager.api.viewMode.value,
+    };
 
-  return {
+    const controlGroupApi = controlGroupApi$.value;
+    let controlGroupReferences: Reference[] | undefined;
+    if (controlGroupApi) {
+      const { rawState: controlGroupSerializedState, references: extractedReferences } =
+        await controlGroupApi.serializeState();
+      controlGroupReferences = extractedReferences;
+      dashboardState.controlGroupInput = controlGroupSerializedState;
+    }
+
+    return {
+      dashboardState,
+      controlGroupReferences,
+      panelReferences,
+    };
+  }
+
+  const trackOverlayApi = initializeTrackOverlay(trackPanel.setFocusedPanelId);
+
+  // Start animating panel transforms 500 ms after dashboard is created.
+  setTimeout(() => animatePanelTransforms$.next(true), 500);
+
+  const dashboardApi = {
+    ...viewModeManager.api,
+    ...dataLoadingManager.api,
+    ...dataViewsManager.api,
+    ...panelsManager.api,
+    ...settingsManager.api,
     ...trackPanel,
-    ...initializeTrackOverlay(trackPanel.setFocusedPanelId),
-    ...initializeUnsavedChanges(
-      initialComponentState.anyMigrationRun,
-      initialComponentState.lastSavedInput
-    ),
-    animatePanelTransforms$,
+    ...unifiedSearchManager.api,
+    ...unsavedChangesManager.api,
+    ...trackOverlayApi,
+    ...initializeTrackContentfulRender(),
+    controlGroupApi$,
+    executionContext: {
+      type: 'dashboard',
+      description: settingsManager.api.panelTitle.value,
```

Comment:
I wonder how much we need to be tied to the `PublishesPanelDescription` + `PublishesPanelTitle` interfaces for the Dashboard, since it's no longer an embeddable? The naming convention is very much designed for **panels**, so it's a tad bit confusing.... But that is very much just a nit

Reply:
agreed. I think we should remove `panel` prefix from the name and instead just call it `title` and `description`. Really its the embeddable `title` and `description`. Whether that embeddable is a panel or a dashboard is irrelevant.